cfme/tests/containers/test_tags.py

```diff
@@ -0,0 +1,98 @@
+import pytest
+
+from cfme.containers.provider import ContainersProvider
+
+
+pytestmark = [
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.provider([ContainersProvider], scope='function')
+]
+
+container_test_items = [
+    ('container_provider'),
+    ('container_projects'),
+    ('container_routes'),
+    ('container_services'),
+    ('container_replicators'),
+    ('container_pods'),
+    ('container_nodes'),
+    ('container_volumes'),
+    ('container_image_registries'),
+    ('container_images'),
+    ('container_templates')
+]
+
+
+def get_collection_entity(appliance, collection_name, provider):
+    if collection_name in ['container_provider']:
+        return provider
+    item_collection = getattr(appliance.collections, collection_name)
+    all_entities = item_collection.all()
+    if all_entities:
+        selected_entity = all_entities[0]
+    else:
+        pytest.skip("No content found for test")
+    for klass in [item_collection]:
+        d = {}
+        ref = {'name': getattr(selected_entity, 'name', None),
+               'project_name': getattr(selected_entity, 'project_name', None),
+               'host': getattr(selected_entity, 'host', None),
+               'id': getattr(selected_entity, 'id', None),
+               'provider': provider}
+        for arg in ['name', 'project_name', 'host', 'id', 'provider']:
+            if arg in [att.name for att in klass.ENTITY.__attrs_attrs__]:
+                d.update({arg: ref[arg]})
+        return item_collection.instantiate(**d)
+
+
+def _tag_cleanup(test_item, tag):
+    tags = test_item.get_tags()
+    if tags:
+        result = (
+            not object_tags.category.display_name == tag.category.display_name and
+            not object_tags.display_name == tag.display_name for object_tags in tags
+        )
+        if not result:
```

Comment:
__REQUIRED__
Is it possible that you forgot `any` or `all` in the `result`? now `result` is generator and always `True`
```

Reply:
result is always `True`, reason (x for x in <something>) is generator, but probably you wanted to do some operation on it, like `all` or `any`.